tests/tls_e2e/common/utility.cpp

```diff
@@ -63,256 +63,82 @@ oe_result_t generate_key_pair(
-    // identity field.
-
-    OE_TRACE_INFO("modulus_size=%ld\n", modulus_size);
-    if (sha256(modulus, modulus_size, signer) != 0)
-    {
-        OE_TRACE_ERROR("sha256 failed\n");
+            "identity->security_version checking failed (%d)\n",
+            identity->security_version);
         goto done;
     }
 
-    if (memcmp(signer, signer_id_buf, signer_id_buf_size) != 0)
-    {
-        OE_TRACE_ERROR("mrsigner is not equal!\n");
-        for (size_t i = 0; i < signer_id_buf_size; i++)
-        {
-            OE_TRACE_ERROR(
-                "0x%x - 0x%x\n", (uint8_t)signer[i], (uint8_t)signer_id_buf[i]);
-        }
-        goto done;
-    }
-    ret = true;
+    // Dump an enclave's unique ID, signer ID and Product ID. They are
+    // MRENCLAVE, MRSIGNER and ISVPRODID for SGX enclaves. In a real scenario,
+    // custom id checking should be done here
+    OE_TRACE_INFO("\nidentity->unique_id :\n");
+    for (int i = 0; i < OE_UNIQUE_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->unique_id[i]);
+
+    OE_TRACE_INFO("\nparsed_report->identity.signer_id :\n");
+    for (int i = 0; i < OE_SIGNER_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    // On a real enclave product, this is the place to check again the enclave
+    // signing key by calling function like verify_mrsigner below. However, we
+    // are not siging test cases, so this checking will be skipped.
+    // tls_between_enclaves sample will have code show how to dothis checking if
+    // (!verify_mrsigner((char *)OTHER_ENCLAVE_PUBLIC_KEY,
+    //                     sizeof(OTHER_ENCLAVE_PUBLIC_KEY),
+    //                     identity->signer_id,
+    //                     sizeof(identity->signer_id)))
+    // {
+    //     OE_TRACE_ERROR("failed:mrsigner not equal!\n");
+    //     goto done;
+    // }
+
+    OE_TRACE_INFO("\nidentity->product_id :\n");
+    for (int i = 0; i < OE_PRODUCT_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->product_id[i]);
+
+    result = OE_OK;
 done:
-    if (signer)
-        free(signer);
-
-    if (modulus != NULL)
-        free(modulus);
-
-    mbedtls_pk_free(&ctx);
-    return ret;
+    return result;
 }
 
-// If set, the verify callback is called for each certificate in the chain.
-// The verification callback is supposed to return 0 on success. Otherwise, the
-// verification failed.
-// The function should return 0 for anything (including invalid
-// certificates) other than fatal error, as a non-zero return
-// code immediately aborts the verification process.
-// For fatal errors, a specific error code should be used
-// (different from MBEDTLS_ERR_X509_CERT_VERIFY_FAILED which
-// should not be returned at this point), or MBEDTLS_ERR_X509_FATAL_ERROR
-// can be used if no better code is available.
-//
-int cert_verify_callback(
-    void* data,
-    mbedtls_x509_crt* crt,
-    int depth,
-    uint32_t* flags)
+int load_oe_modules()
```

Comment:
Consider use `oe_result_t` instead of `int` as the return type, being more consistent with OE code base (also ones in the openssl_utility.h).
With `oe_result_t`, you can simply use `OE_CHECK(load_oe_modules)` or similar OE test routines.